astropy/nddata/utils.py

```diff
@@ -456,3 +469,267 @@ def block_replicate(data, block_size, conserve_sum=True):
         data = data / float(np.prod(block_size))
 
     return data
+
+
+class Cutout(object):
+    def __init__(self, data, position, shape, wcs=None, mode='trim',
+                 fill_value=np.nan):
+        """
+        Create a cutout object from a 2D array.
+
+        The returned object will contain the 2D cutout array and the
+        updated WCS, if a `~astropy.wcs.WCS` object is input.
```

Comment:
Can you add a sentence or two describing whether this creates a view into `data` or makes a copy of part of it?
I.e. does the cutout object have references back to `data` (or `wcs`) and can / should I use it to modify that part of the `data` array?

One application @adonath and I have for cutouts is that we have a survey image.
Then we create cutouts of overlapping groups of sources (currently shelling out and calling `ftcopy` to make sure the FITS header is updated correctly :disappointed: ) and fit models on those.
Then  at the end we copy the small model images back into a survey model image.

I guess my question is: is the copying back into the survey image possible / easy with what's implemented here?